### PR TITLE
Handle pre-Homestead contract creation when not enough gas

### PR DESCRIPTION
**Pull request by LukasGasior1**
We need this logic in two places: CREATE op code (already there) and Ledger (this PR).

### CODE REVIEW

File: src/main/scala/io/iohk/ethereum/ledger/Ledger.scala [hunk 1]
```diff
@@ -317,13 +317,16 @@ class Ledger(vm: VM) extends Logger {
 
   private def saveNewContract(address: Address, result: PR, config: EvmConfig): PR = {
```
Comment (AlanVerbner) [inline src/main/scala/io/iohk/ethereum/ledger/Ledger.scala]: The code looks good and i'm still running to see what block do I get. Do you think it's too much trouble to create a test to verify this if branches are ok?

**Comment by AlanVerbner** [comment]
@ntallar there is a FIXME message in Ledger line 116, is it related to this fix?

**Comment by LukasGasior1** [comment]
It seems related. I don't think there are any other changes in Homestead with regard to error handling

**Comment by ntallar** [comment]
It is related, it was just a remainder that the fix from this PR was needed, so it can be removed :+1:

**Comment by ntallar** [comment]
Apart from @AlanVerbner's comment on having tests for this, LGTM!

File: src/main/scala/io/iohk/ethereum/ledger/Ledger.scala (hunk 1 above)
Comment (LukasGasior1) [inline src/main/scala/io/iohk/ethereum/ledger/Ledger.scala]: @AlanVerbner I've added a test

**Pull request merged**